relay-server/src/actors/project.rs

```diff
@@ -1215,57 +1243,90 @@ impl Message for FetchProjectState {
 }
 
 impl Handler<FetchProjectState> for ProjectCache {
-    type Result = Response<ProjectStateResponse, ()>;
+    type Result = ResponseActFuture<Self, ProjectStateResponse, ()>;
```

Comment:
It's a bit unfortunate that we're losing the optimization with the synchronous response here. Also it makes the function implementation quite a bit more ugly.

After talking offline a bit, there's no way around this if we want to keep accessing the cache from the `ProjectManager` actor. Otherwise we would have to pass the cache into the `Project` actor, which would technically be a slightly more correct solution, but requires yet another refactor. 

That said, having a synchronous local response is not that much of an optimization, as it only affects local configuration, which is an edge case.